packages/@uppy/transloadit/src/index.js

```diff
@@ -191,6 +191,15 @@ export default class Transloadit extends BasePlugin {
       expectedFiles: fileIDs.length,
       signature: options.signature,
     }).then((newAssembly) => {
+      const files = this.uppy.getFiles().filter(({ id }) => fileIDs.includes(id))
+      if (files.length !== fileIDs.length) {
+        if (files.length === 0) {
+          // All files have been removed, cancelling.
```

Comment:
I am confused by this. What are we trying to achieve, why check for files when creating an assembly? “cancelling” — what are we canceling, it just returns null?

Reply:
See the next line, we cancel the Assembly (https://github.com/transloadit/uppy/pull/3893/files#diff-5d92711b94b1d7e763474c1f9b32451a247a93597351cc18db9a8320996065f9R198).

Reply:
> What are we trying to achieve?

One of our paying customer reported that when removing all files programatically (instead of using `cancelAll`), the assembly would keep polling its status, waiting for the files to be uploaded. Checking for the files ensures the assembly is cancelled and resolve the unnecessary polling issue.

Reply:
Yeah, I know the issue, was confused about the check happening on assembly creation and not during execution, all clear now.